python-sdk/nuscenes/utils/map_mask.py

```diff
@@ -14,109 +16,104 @@
-        return self._transf_matrix
-
-    def dilate_mask(self, dist_thresh: float = 2.0) -> None:
-        """
-        Dilates the mask by a distance threshold.
-        :param dist_thresh: This parameter specifies the threshold on the distance from the semantic prior mask.
-            The semantic prior mask is dilated to include points which are within this distance from itself.
-        """
-        # Distance to nearest foreground in mask.
-        distance_mask = cv2.distanceTransform((self.foreground - self.mask).astype(np.uint8), cv2.DIST_L2, 5)
-        distance_mask = (distance_mask * self.precision).astype(np.float32)
-
-        self._mask = (distance_mask < dist_thresh).astype(np.uint8) * self.foreground
-
-    def export_to_png(self, filename: str='mask.png') -> None:
-        """
-        Export mask to png file.
-        :param filename: Path to the png file.
-        """
-        cv2.imwrite(filename=filename, img=self.mask)
+        return np.array([[1.0 / self.resolution, 0, 0, 0],
+                         [0, -1.0 / self.resolution, 0, self._base_mask.shape[0]],
+                         [0, 0, 1, 0], [0, 0, 0, 1]])
 
-    def is_on_mask(self, x, y) -> np.array:
+    def is_on_mask(self, x, y, dilation=0) -> np.array:
         """
-        Determine whether the points are on binary mask.
+        Determine whether the given coordinates are on the (optionally dilated) map mask.
         :param x: Global x coordinates. Can be a scalar, list or a numpy array of x coordinates.
         :param y: Global y coordinates. Can be a scalar, list or a numpy array of x coordinates.
+        :param dilation: <float>. Optional dilation of map mask.
         :return: <np.bool: x.shape>. Whether the points are on the mask.
         """
+        px, py = self.to_map_coord(x, y)
 
-        if not isinstance(x, np.ndarray):
-            x = np.array(x)
-        if not isinstance(y, np.ndarray):
-            y = np.array(y)
-
-        assert x.shape == y.shape
-
-        if x.ndim == 0:
-            x = np.atleast_1d(x)
-        if y.ndim == 0:
-            y = np.atleast_1d(y)
-
-        assert x.ndim == y.ndim == 1
-
-        px, py = self.get_pixel(x, y)
-
-        on_mask = np.ones(x.size, dtype=np.bool)
+        on_mask = np.ones(px.size, dtype=np.bool)
+        this_mask = self.mask(dilation)
 
         on_mask[px < 0] = False
-        on_mask[px >= self.mask.shape[1]] = False
+        on_mask[px >= this_mask.shape[1]] = False
         on_mask[py < 0] = False
-        on_mask[py >= self.mask.shape[0]] = False
+        on_mask[py >= this_mask.shape[0]] = False
 
-        on_mask[on_mask] = (self.mask[py[on_mask], px[on_mask]] == self.foreground)
+        on_mask[on_mask] = this_mask[py[on_mask], px[on_mask]] == self.foreground
 
         return on_mask
 
-    def get_pixel(self, x: np.array, y: np.array) -> Tuple[np.ndarray, np.ndarray]:
+    def to_map_coord(self, x, y) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
The name should be updated. It actually returns the pixel coordinates in the map image and not the global map coordinates. The docstring should be updated as well.